src/cargo/ops/cargo_package.rs

```diff
@@ -354,6 +316,12 @@ impl LocalDependencies {
             .map(|name| self.packages[&name].clone())
             .collect()
     }
+
+    pub fn has_no_dependencies(&self) -> bool {
```

Comment:
This commit seems to have multiple refactors in it, not just moving of a function.  I'm having a hard time following what each change is intended to do.  Could you split this further up?

Reply:
I've split the second commit into a pure move, followed by some changes. "Refactor" was probably the wrong word; there are a few logic changes because I wanted to reuse `publish`'s `get_source_id` logic for `package`.